lib/elixir/src/elixir_with.erl

```diff
@@ -55,11 +55,14 @@ expand({'<-', Meta, [Left, Right]}, E) ->
 expand(X, E) ->
   elixir_exp:expand(X, E).
 
-expand_else(KV, E) when is_list(KV) ->
+expand_else(_Meta, KV, E) when is_list(KV) ->
   {[{do, EClauses}], EC} = elixir_exp_clauses:'case'([], [{do, KV}], E),
   {[{else, EClauses}], EC};
-expand_else(nil, E) ->
-  {[], E}.
+expand_else(_Meta, nil, E) ->
+  {[], E};
+expand_else(Meta, _KV, E) ->
+  elixir_errors:compile_error(Meta, ?m(E, file),
+  "expected -> clauses in else block", []).
```

Comment:
Thank you for the feedback.
@whatyouhide Oh, you're right, my indentation is weird, I'll fix it with the message!
I agree with @michalmuskala, keeping the same message as for `case` is more consistent.

The case message is

> expected -> clauses for do in case

so what do you think about about

> expected -> clauses for else in with